source/src/protocols/motifs/LigandDiscoverySearch.cc

```diff
@@ -292,694 +366,1790 @@ utility::vector1<core::conformation::ResidueOP> LigandDiscoverySearch::get_all_r
+{
+	fa_rep_fxn_->score(*minipose);
 
-	//hold the number of placements that pass all filters and could enter the top 100 placements
-	int passed_placement_counter = 0;
+	//high fa_rep means clashing, want low fa_rep
+	fa_rep = minipose->energies().residue_total_energies(minipose->size())[core::scoring::fa_rep];
 
-	//now we have the filtered motif library to work with, run through each  atom trio in each ligand and try to match it against all motifs for the residue
-	for ( core::Size tracker = 1; tracker <= all_residues_.size(); ++tracker ) {
+	//check if fa_rep is good
+	//positive score is bad
+	//best scores are negative and closest to 0
+	if ( fa_rep > fa_rep_cutoff_ ) {
+		return false;
+	}
 
-		bool ligand_added = false;
+	fa_atr_fxn_->score(*minipose);
 
-		//core::chemical::ResidueTypeCOP ligres(ref);
-		//convert ligres to be a ResidueOP type
-		core::conformation::ResidueOP ligresOP = all_residues_[tracker];
+	fa_atr = minipose->energies().residue_total_energies(minipose->size())[core::scoring::fa_atr];
 
-		ms_tr << "On ligand " << ligresOP->name() << std::endl;
+	//run fa_atr check
+	//don't keep if fa_atr is greater than cutoff
+	if ( fa_atr > fa_atr_cutoff_ ) {
+		return false;
+	}
 
-		const core::Real lig_nbr_radius = ligresOP->nbr_radius();
+	//score whole minipose with atr_rep function
+	fa_atr_rep_score_before = fa_atr_rep_fxn_->score(*minipose);
 
-		ms_tr << "NBR_RADIUS of ligand is: " << lig_nbr_radius << std::endl;
+	//check if worse than cutoff
+	if ( fa_atr_rep_score_before > fa_atr_rep_cutoff_ ) {
+		return false;
+	}
 
-		int ligand_passing_counter  = 0;
-		int ligand_clashing_counter = 0;
+	return true;
+}
 
-		//derive a mutable residue type, used in high res docker
-		core::chemical::MutableResidueTypeOP lig_mrt( new core::chemical::MutableResidueType( ligresOP->type() ) );
+// @brief create a constraint set on the 3 ligand motif atoms (the last residue in working_pose_) to reduce their movement before applying a highresdock
+void LigandDiscoverySearch::add_constraints_to_working_pose(const core::Size trip_atom_1, const core::Size trip_atom_2, const core::Size trip_atom_3, const core::Size working_position, const core::conformation::ResidueOP ligresOP)
+{
+	//create constraints for ligand wiggling to add to the pose
+	constraints::ConstraintSetOP sc_cst_set( new constraints::ConstraintSet() );
 
-		// This is to make an atomtypeset to get atomtype integers
-		core::chemical::AtomTypeSetCOP atset = core::chemical::ChemicalManager::get_instance()->atom_type_set( FA_STANDARD );
+	core::scoring::func::FuncOP fx1( new core::scoring::func::HarmonicFunc( 0.0, 1.0 ) );
+	sc_cst_set->add_constraint( core::scoring::constraints::ConstraintCOP( utility::pointer::make_shared< core::scoring::constraints::CoordinateConstraint >( core::id::AtomID( trip_atom_1, working_pose_->size() ), core::id::AtomID( working_pose_->residue( working_position ).atom_index( "CA" ), 1 ), ligresOP->xyz( trip_atom_1 ), fx1 ) ) );
```

Comment:
`utility::pointer::make_shared<core::scoring::constraints::CoordinateConstraint>` will create a new OP of the type `core::scoring::constraints::CoordinateConstraintOP`, you don't have to manually cast it to `core::scoring::constraints::ConstraintCOP` -- it should be automatically convertable to the base-class OP, and the OP -> COP transition should also happen automatically.